### PR TITLE
Add decorate to all contents types and added text base error response

### LINKED ISSUE #42870: Decorated stacktraces in dev mode are not provided when using tools like curl or postman
### Describe the bug

Decorated stacktraces in dev mode are not provided when using tools like curl or postman
Decorated stacktraces were introduced in https://github.com/quarkusio/quarkus/pull/37034 /cc @geoand 

When using browser I can see the code example
Screenshot 2024-08-29 at 16 35 15

Postman / curl response doesn't have these details
Screenshot 2024-08-29 at 16 34 40

I'm filing this as a bug because I think the details provided by the "error" page should be consistent.
Feel free to change this to enhancement if you have an opinion about that.

### Expected behavior

Decorated stacktraces in dev mode are also provided when using tools like curl or postman

### Actual behavior

Decorated stacktraces in dev mode are not provided when using tools like curl or postman

### How to Reproduce?

 - Get app from https://code.quarkus.io/
 - Run it in dev mode
 - Adjust `GreetingResource` to through NPE, e.g. `String foo = null; return "Hello from Quarkus REST" + foo.length();`
 - Hit http://localhost:8080/hello in postman or curl

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 21

### Quarkus version or git rev

Quarkus 3.14.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by phillip-kruger**
Fix #42870

I also aligned this with the 404 page (see https://github.com/quarkusio/quarkus/pull/41989) by adding a text based response for curl.

curl_error

### CODE REVIEW

**Comment by phillip-kruger** [comment]
CI failures are related. I'll have a look

**Comment by rsvoboda** [comment]
OK, I will check the this once you update the PR @phillip-kruger

**Comment by phillip-kruger** [comment]
Ok @rsvoboda - please keep an eye on CI. I'll check in again in the morning.

**Comment by sberyozkin** [comment]
@phillip-kruger Hi Phillip, I'm not seeing in this PR where a devmode check is done, or did I miss it ?

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java [hunk 1]
```diff
@@ -181,30 +190,94 @@ public void accept(Throwable throwable) {
...
+        try (StringWriter sw = new StringWriter()) {
+            sw.write(NL + HEADING + NL);
+            sw.write("------------------------" + NL);
+            sw.write(NL);
+            sw.write("Details:");
+            sw.write(NL);
+            sw.write(TAB + details);
+            sw.write(NL);
+            if (decoratedString != null) {
+                sw.write("Decorate:");
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java:248]: It's decorate in the config option ?  @maxandersen @geoand - wdyt ?

[the remark this answers is not in the document]
Comment (phillip-kruger) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java]: I made it `Decorate (Source Code)` for now. Let me know what it should be.

**Comment by phillip-kruger** [comment]
@sberyozkin - this is existing functionality, hence not in this PR

**Review by geoand — APPROVED** [review]
Nice!

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java [hunk 2]
```diff
@@ -306,23 +379,41 @@ private ContentTypes() {
...
+        }
 
         static String pickFirstSupportedAndAcceptedContentType(RoutingContext context) {
             List<MIMEHeader> acceptableTypes = context.parsedHeaders().accept();
-            MIMEHeader result = context.parsedHeaders().findBestUserAcceptedIn(acceptableTypes, SUPPORTED);
-            return result == null ? null : result.value();
+
+            String userAgent = context.request().getHeader("User-Agent");
+            if (userAgent != null && (userAgent.toLowerCase(Locale.ROOT).startsWith("wget/")
+                    || userAgent.toLowerCase(Locale.ROOT).startsWith("curl/"))) {
```
Comment (geoand) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java:410]: At some point we should make this a little nice so we can easily add more values. 
I am thinking of the following HTTP clients that I personally use:

`httpie`, `IntelliJ HTTP Client` and `Postman` (although I am not sure what values they use for the header - if any)

**Comment by phillip-kruger** [comment]
We can maybe move this to a config ? With these as defaults?

**Comment by geoand** [comment]
I suppose

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**